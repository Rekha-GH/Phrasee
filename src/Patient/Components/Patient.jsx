import React from 'react';
import { getDateDMMMMYYYY } from '../../Common/Utils/DateUtils';

const Patient = React.memo(({ name, joined, last_visit_date, is_completed }) => {
    return (
        <div className={'patient-item'}>
            name: {name}<br />
            joined: {getDateDMMMMYYYY(joined)}<br />
            Last visit: {getDateDMMMMYYYY(last_visit_date)}<br />
            Completed: {is_completed ? 'yes' : 'no'}
        </div>
    );
});

Patient.displayName = 'Patient';
export default Patient;