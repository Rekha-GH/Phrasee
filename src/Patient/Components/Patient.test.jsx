import React from 'react';
import toJson from 'enzyme-to-json';
import * as ReactRedux from 'react-redux';


import Patient from './Patient';
import { shallow } from '../../setupTests';

describe('Patient Tests', () => {
  test('Patient renders correctly', () => {
    const wrapper = shallow(<Patient
        is_completed={false}
        joined={"2020-08-06T07:54:00.232Z"}
        last_visit_date={"2020-08-06T00:00:00.000Z"}
        name={"Adam S"}
    />);
    
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})
