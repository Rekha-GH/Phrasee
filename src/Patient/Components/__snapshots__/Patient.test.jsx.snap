// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Patient Tests Patient renders correctly 1`] = `
<div
  className="patient-item"
>
  name: 
  Adam S
  <br />
  joined: 
  6 August 2020
  <br />
  Last visit: 
  6 August 2020
  <br />
  Completed: 
  no
</div>
`;
